### PR TITLE
bugfix for writing genpept with long ids.

### LINKED ISSUE #3732: Can't write peptide genbank files with extra long headers
This line
https://github.com/biopython/biopython/blob/cb4f6edf59f9434c177dcd1af21dd588f9dfa161/Bio/SeqIO/InsdcIO.py#L799

should be changed to:

```
splitline[3].strip() == "aa"
```

because in genbank peptide files there isn't a molecule type, so that field gets eaten by the split and it won't ever be `""`

### PR DESCRIPTION
**Pull request by seanrjohnson**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #3732 

One line change as suggested in the issue post, plus a test.

### CODE REVIEW

**Review by mdehoon — CHANGES_REQUESTED** [review]
Instead of commenting out lines, just remove them.

**Comment by peterjc** [comment]
Why are you removing all those warnings anyway?

**Comment by seanrjohnson** [comment]
> Why are you removing all those warnings anyway?

That's a good question. Actually I didn't mean those last few commits to be part of the pull request (didn't realize, or maybe forgot, that Github updates pull requests with later commits to the same branch, I should have made a new branch for those).

But to answer your question, I was trying to optimize the speed of the genbank parser, so I did profiling and found that those lines cause major performance hits and check for very rare conditions.

I'll try to revert the branch back to the commit that I intended to include.

**Comment by peterjc** [comment]
Yes, updating the branch updates the pull request too. Very useful once you know to expect this.

I wondered if the de-quoting change from regex to plain Python would be good or bad from a performance point of view. Avoiding the regex would be easier to read.

**Comment by seanrjohnson** [comment]
avoiding the regexes is helpful for speed. The changes in that commit are all the result of profiling I did. All together the changes almost halve the parsing time (9 seconds to 5 seconds for the example I was profiling on).

**Comment by peterjc** [comment]
It has been years since I last profiled the GenBank parser, I guess we've built up some additional checks which come at a speed cost. Hopefully we can speed them up now you've flagged some hotspots.

**Comment by seanrjohnson** [comment]
I made a commit reverting changes irrelevant for this pull request (except the one changing the quotes regex to checking the first and last characters). Sorry for the confusion.

**Comment by peterjc** [comment]
Note you've accidentally deleted the symlink ``LICENSE`` here.

Feel free to clean up the branch history (e.g. ``git rebase`` into a single commit), but if not we can do a squash-and-merge of this pull request instead.

**Comment by seanrjohnson** [comment]
Ok, I tried. The history looks even more complicated now, so not sure if I did it right.

**Comment by peterjc** [comment]
The history looks cleaner now (no merges). If you're not familiar with git-rebase, it isn't immediately obvious how to use it to combine commits and re-write the commit messages. Don't worry about it.

We can squash-and-merge with a commit comment about speeding up the quote removal, and addressing the false positive warning for #3732.

File: Bio/GenBank/__init__.py [hunk 1]
```diff
@@ -1285,7 +1285,8 @@ def feature_qualifier(self, key, value):
             return
 
         # Remove enclosing quotation marks
-        value = re.sub('^"|"$', "", value)
+        if (value[0] == '"') and (value[-1] == '"'):
```
[the remark this answers is not in the document]
Comment (seanrjohnson) [reply Bio/GenBank/__init__.py:1288]: I suppose `if len(value) > 1 and value[0] == '"' and value[-1] == '"'`
would address all of those concerns without new weird side effects?

[the remark this answers is not in the document]
Comment (peterjc) [reply Bio/GenBank/__init__.py:1288]: Pythonic style would be checking non-zero ``value`` by treating it as a boolean: ``if value and value[0] == '"' and value[-1] == '"':``

But yes, that's the simplest/safest change I could come up with too.

[the remark this answers is not in the document]
Comment (peterjc) [reply Bio/GenBank/__init__.py:1288]: P.S. I'm pleased to see the test suite caught this, ``IndexError: string index out of range``

[the remark this answers is not in the document]
Comment (seanrjohnson) [reply Bio/GenBank/__init__.py]: ok, I made that change.

[the remark this answers is not in the document]
Comment (seanrjohnson) [reply Bio/GenBank/__init__.py]: @peterjc 
I just ran the benchmark again, and just that one change still seems to speed it up a bit.
My test file takes 9.1 seconds to parse with the `value = re.sub('^"|"$', "", value)`. but when I change to `if len(value) > 1 and value[0] == '"' and value[-1] == '"'` it goes down to 7.7 seconds.

[the remark this answers is not in the document]
Comment (peterjc) [reply Bio/GenBank/__init__.py]: Nice 🚀

**Pull request merged**